### ISSUE TITLE
Request failed with status code 400.

### ISSUE THREAD

**Issue by affan00** [evercam/evercam-dashboardv2#354]
Sometimes API request failed with status code 400, We need to fix this ASAP.
![image](https://user-images.githubusercontent.com/14149453/70512891-5aa12580-1b52-11ea-86e9-ae1f0e7c643e.png)

**Comment by ijunaid8989** [comment]
This issue seems to belong both sides, 
1. Dashboard
2. API.

I have a hunch that the dashboard is sending more frequent requests to the server than usual,

```
 35550  89.100.74.26 - - [13/Nov/2017:09:00:58 +0100] "GET /v1/cameras/ponds-killruddery/live/snapshot?api_id=xxx&api_key=xxxxxx HTTP/1.1" 500 41 "-" "Mozilla/3.0 (compatible; Indy Library)"
 35552  89.100.74.26 - - [13/Nov/2017:09:00:58 +0100] "GET /v1/cameras/ponds-killruddery/live/snapshot?api_id=xxx&api_key=xxxxxx HTTP/1.1" 500 41 "-" "Mozilla/3.0 (compatible; Indy Library)"
 35558  89.100.74.26 - - [13/Nov/2017:09:01:07 +0100] "GET /v1/cameras/ponds-killruddery/live/snapshot?api_id=xxx&api_key=xxxxxx HTTP/1.1" 500 41 "-" "Mozilla/3.0 (compatible; Indy Library)"
 35560  89.100.74.26 - - [13/Nov/2017:09:01:08 +0100] "GET /v1/cameras/ponds-killruddery/live/snapshot?api_id=xxx&api_key=xxxxxx HTTP/1.1" 500 41 "-" "Mozilla/3.0 (compatible; Indy Library)"
 35566  89.100.74.26 - - [13/Nov/2017:09:01:18 +0100] "GET /v1/cameras/ponds-killruddery/live/snapshot?api_id=xxx&api_key=xxxxxx HTTP/1.1" 500 41 "-" "Mozilla/3.0 (compatible; Indy Library)"
 35568  89.100.74.26 - - [13/Nov/2017:09:01:18 +0100] "GET /v1/cameras/ponds-killruddery/live/snapshot?api_id=xxx&api_key=xxxxxx HTTP/1.1" 500 41 "-" "Mozilla/3.0 (compatible; Indy Library)"
 35650  89.100.74.26 - - [13/Nov/2017:09:01:38 +0100] "GET /v1/cameras/ponds-killruddery/live/snapshot?api_id=xxx&api_key=xxxxxx HTTP/1.1" 500 41 "-" "Mozilla/3.0 (compatible; Indy Library)"


 35780  89.100.74.26 - - [13/Nov/2017:09:02:38 +0100] "GET /v1/cameras/ponds-killruddery/live/snapshot?api_id=xxx&api_key=xxxxxx HTTP/1.1" 500 41 "-" "Mozilla/3.0 (compatible; Indy Library)"
 35861  89.100.74.26 - - [13/Nov/2017:09:02:48 +0100] "GET /v1/cameras/ponds-killruddery/live/snapshot?api_id=xxx&api_key=xxxxxx HTTP/1.1" 500 41 "-" "Mozilla/3.0 (compatible; Indy Library)"
 35862  89.100.74.26 - - [13/Nov/2017:09:02:48 +0100] "GET /v1/cameras/gateout/live/snapshot?api_id=xxx&api_key=xxxxxx HTTP/1.1" 500 41 "-" "Mozilla/3.0 (compatible; Indy Library)"
 35885  89.100.74.26 - - [13/Nov/2017:09:02:58 +0100] "GET /v1/cameras/ponds-killruddery/live/snapshot?api_id=xxx&api_key=xxxxxx HTTP/1.1" 500 41 "-" "Mozilla/3.0 (compatible; Indy Library)"
 35890  89.100.74.26 - - [13/Nov/2017:09:03:08 +0100] "GET /v1/cameras/ponds-killruddery/live/snapshot?api_id=xxx&api_key=xxxxxx HTTP/1.1" 500 41 "-" "Mozilla/3.0 (compatible; Indy Library)"
 35892  89.100.74.26 - - [13/Nov/2017:09:03:08 +0100] "GET /v1/cameras/ponds-killruddery/live/snapshot?api_id=xxx&api_key=xxxxxx HTTP/1.1" 500 41 "-" "Mozilla/3.0 (compatible; Indy Library)"
 35917  89.100.74.26 - - [13/Nov/2017:09:03:18 +0100] "GET /v1/cameras/ponds-killruddery/live/snapshot?api_id=xxx&api_key=xxxxxx HTTP/1.1" 500 41 "-" "Mozilla/3.0 (compatible; Indy Library)"
 35921  89.100.74.26 - - [13/Nov/2017:09:03:18 +0100] "GET /v1/cameras/ponds-killruddery/live/snapshot?api_id=xxx&api_key=xxxxxx HTTP/1.1" 500 41 "-" "Mozilla/3.0 (compatible; Indy Library)"
```
for above logs you need to do `cat -n /usr/local/nginx/logs/access.log | grep 500`

and these requests are not going to Phoenix server but being sent 500 from nginx. 

Right now there is no way to know when this is happening because sumo logic is connected with Phoenix logs not nginx and I think it would be better if we also connect `/usr/local/nginx/logs/access.log` log file with sumo to check for 500 error and then we can get notified how many times this happens.

also 

```
019-12-11 12:33:12.157 request_id=Fd9N2bvVQU_13bsB4Z8C [info] GET /v2/cameras/103co-pizue/recordings/snapshots/NaN/NaN/NaN/NaN
2019-12-11 12:33:12.288 request_id=Fd9N2bvVQU_13bsB4Z8C [error] [Handled Errors] [GET /v2/cameras/103co-pizue/recordings/snapshots/NaN/NaN/NaN/NaN] [] [construction@evercam.io] [89.100.74.80] [400] ["The parameter 'year' isn't valid."]
2019-12-11 12:33:12.288 request_id=Fd9N2bvVQU_13bsB4Z8C [info] Sent 400 in 130ms
2019-12-11 12:33:12.346 request_id=Fd9N2ccRHN72WLMBqN2F [info] GET /v2/cameras/103co-pizue/recordings/snapshots/Invalid%20date/Invalid%20date/days
2019-12-11 12:33:12.475 request_id=Fd9N2ccRHN72WLMBqN2F [error] [Handled Errors] [GET /v2/cameras/103co-pizue/recordings/snapshots/Invalid%20date/Invalid%20date/days] [] [construction@evercam.io] [89.100.74.80] [400] ["The parameter 'year' isn't valid."]
2019-12-11 12:33:12.475 request_id=Fd9N2ccRHN72WLMBqN2F [info] Sent 400 in 129ms
2019-12-11 12:33:12.547 request_id=Fd9N2dMMJ-hxhZ8B4Z-i [info] GET /v2/cameras/103co-pizue/recordings/snapshots/Invalid%20date/Invalid%20date/Invalid%20date/hours
2019-12-11 12:33:12.676 request_id=Fd9N2dMMJ-hxhZ8B4Z-i [error] [Handled Errors] [GET /v2/cameras/103co-pizue/recordings/snapshots/Invalid%20date/Invalid%20date/Invalid%20date/hours] [] [construction@evercam.io] [89.100.74.80] [400] ["The parameter 'year' isn't valid."]
````

**Comment by ijunaid8989** [comment]
After tracing back to Nginx and its log files. it seems like we were having those errors.
```
failed (24: Too many open files) while connecting to upstream,
```

which meant to be ulimit and FDs are reaching their limits.

In systemd for evercam server.

```
root@Ubuntu-1704-zesty-64-minimal:~# cat /etc/systemd/system/evercam_media.service
[Unit]
Description=Evercam Media
After=network.target

[Service]
Type=forking
LimitNOFILE=1000000
User=root
```
it was all fine but on Nginx conf, we haven't defined anything. we the help of this, https://stackoverflow.com/questions/46172986/nginx-loadbalancer-too-many-open-files

I increased `worker_rlimit_nofile    20000;` which seems to push away the above error.

Then after a while In logs, the errors start to come as 

`1024 worker_connections are not enough`

1024 is default value for worker connections. According to this, https://stackoverflow.com/questions/56723449/nginx-error-1024-worker-connections-are-not-enough
 I increased them to 4096.

Right now, everything seems to work fine. But it seems like there is something "extra" we are doing in dashboardv2 which was not happening in v1. So we should look for it, as we can change and increase things on the evercam server-side to some extent.

I am trying to link Sumo with Nginx for further details and errors.
